src/contexts/price-oracle.context.tsx

```diff
@@ -31,57 +55,89 @@ const priceOracleContext = createContext<PriceOracleContext>({
-  const [quoterContract, setQuoterContract] = useState<UniswapQuoter>();
-
-  const changePreferredCurrency = useCallback((currency: Currency): void => {
-    storage.setCurrency(currency);
-    setPreferredCurrency(currency);
-  }, []);
+  const [uniswapV2Router02Contract, setUniswapV2Router02Contract] = useState<UniswapV2Router02>();
 
   const getTokenPrice = useCallback(
-    async (tokenAddress: string): Promise<number> => {
+    async ({ token, chain }: GetTokenPriceParams): Promise<BigNumber> => {
+      const erc20Token =
+        token.address === ethers.constants.AddressZero
+          ? getChainTokens(chain).find((t) => t.symbol === "WETH")
+          : token;
+
+      if (!erc20Token) {
+        throw new Error(
+          `ETH is not a valid ERC-20 token and its wrapped version "WETH" could not be found on the ${getChainName(
+            chain
+          )} chain`
+        );
+      }
       if (env === undefined) {
         throw new Error("Env is not available");
       }
 
-      if (quoterContract === undefined) {
-        throw new Error("Price oracle contract is not available");
+      if (uniswapV2Router02Contract === undefined) {
+        throw new Error("Uniswap V2 Router02 contract is not available");
       }
 
       if (fiatExchangeRates === undefined) {
         throw new Error("Fiat exchange rates are not available");
       }
 
-      const rate = await quoterContract.callStatic.quoteExactInputSingle(
-        tokenAddress,
-        env.fiatExchangeRates.usdtToken.address,
-        UNISWAP_V3_POOL_FEE,
-        parseUnits("1"),
-        0
+      const ethereumChain = env.chains.find((chain) => chain.key === "ethereum");
+
+      if (ethereumChain === undefined) {
+        throw new Error("Ethereum chain is not available");
+      }
+
+      const uniswapPairContractAddress = computePairAddress({
+        tokenA: erc20Token,
+        tokenB: env.fiatExchangeRates.usdcToken,
+      });
+
+      const uniswapPairContract = UniswapV2Pair__factory.connect(
+        uniswapPairContractAddress,
+        ethereumChain.provider
       );
-      const usdPrice = Number(formatUnits(rate, env.fiatExchangeRates.usdtToken.decimals));
-      const fiatExchangeRate = fiatExchangeRates[preferredCurrency];
 
-      if (!fiatExchangeRate) {
+      const [reserveIn, reserveOut] = await uniswapPairContract.callStatic.getReserves();
+
+      const rate = await uniswapV2Router02Contract.callStatic.getAmountOut(
+        parseUnits("1", erc20Token.decimals),
+        reserveIn,
+        reserveOut
+      );
+
+      const fiatExchangeRate = fiatExchangeRates[getCurrency()];
+
+      if (fiatExchangeRate === undefined) {
         throw new Error("Fiat exchange rate not found");
       }
 
-      return usdPrice * fiatExchangeRate;
+      const price = multiplyAmounts(
+        { value: rate, precision: env.fiatExchangeRates.usdcToken.decimals },
+        {
+          value: fiatStringToBigNumber(fiatExchangeRate.toString()),
```

Comment:
This function seems to be only used here, can't we just use `parseUnits(fiatExchangeRate.toString(), PREFERRED_CURRENCY_ARITHMETIC_PRECISION);`.

Reply:
Agreed. The functions `fiatStringToBigNumber` and `fiatBigNumberToString` have been removed.